core/sys/process.h

```diff
@@ -529,3 +529,26 @@ CCIF extern struct process *process_list;
 
 /** @} */
 /** @} */
+ //shaves off any leading  space 
+char* process_space_begin(char *input){
+	while(*input==' '){
+		input++;
+	}
+	return input;
+}
+
+//Find first space in a string, then replace the space with zero.
+void nullterminate(char *cptr){
+ 	for(; *cptr != ' ' && *cptr !=0; ++cptr);
+        *cptr=0;
+}
+
+static int is_broadcast_addr(uint_t mode, uint_t *addr){
+	int i=mode==FRAME802154_SHORTADDRMODE ? 2 : 8;
+	while(i-->0){
+		if(addr[i] != oxff){
+			return 0;
+		}
+	}
+	return 1;
+}
```

Comment:
I'm very sorry that I'm not a Contiki user, so I don't know the code style. I do some works on software static analysis to detect bugs. the same code of function "process_space_begin" is used 11 times in contiki. "nullterminate" is used 4 times. "is_broadcast_addr" is used 3 times.May be these codes can be write as functions in a header file or using some better ways to save storage and improve efficiency.